src/Adapters/Storage/Mongo/MongoStorageAdapter.js

```diff
@@ -474,6 +535,35 @@ export class MongoStorageAdapter {
     return this._adaptiveCollection(className)
       .then(collection => collection._mongoCollection.indexes());
   }
+
+  dropIndex(className, index) {
+    return this._adaptiveCollection(className)
+      .then(collection => collection._mongoCollection.dropIndex(index));
+  }
+
+  dropAllIndexes(className) {
+    return this._adaptiveCollection(className)
+      .then(collection => collection._mongoCollection.dropIndexes());
+  }
+
+  updateSchemaWithIndexes() {
+    return this.getAllClasses()
+      .then((classes) => {
+        const promises = classes.map((schema) => {
+          return this.getIndexes(schema.className).then((indexes) => {
+            indexes = indexes.reduce((obj, index) => {
+              obj[index.name] = index.key;
+              return obj;
+            }, {});
+            return this._schemaCollection()
+              .then(schemaCollection => schemaCollection.updateSchema(schema.className, {
+                $set: { _metadata: { indexes: indexes } }
+              }));
+          });
```

Comment:
@dplewis finally found the problem. When attempting to auto-update the schema with indexes during startup it's possible that the classes you've been returned have _not_ been setup in the DB yet. This is the case for `_User` and `_Role`, which are both built-ins to parse and will be present even if they haven't been truly setup yet.

The tests cause this to trigger an error due to the continual dropping/reading of the collections, while the defaults continue to 'seem' present in the db. 

To handle this all you really need to do is add `catch` block to the promise returned by `this.getIndexes`. 
```js
return this.getIndexes(schema.className).then((indexes) => {
  indexes = indexes.reduce((obj, index) => {
    obj[index.name] = index.key;
    return obj;
  }, {});
  return this._schemaCollection()
    .then(schemaCollection => schemaCollection.updateSchema(schema.className, {
	  $set: { _metadata: { indexes: indexes } }
    }));
}).catch(() => {
  // catch any errors
});
```
The reason we weren't seeing this in postgres is due to `updateSchemaWithIndexes` in the PostgresStorageAdapter doing nothing more than returning a resolved promise.

Reply:
Thanks for looking into this. I'll fix those changes later today.

Reply:
Whenever you get a moment, no rush.

Reply:
I cleaned up the PR. I have the changes you requested. How do we update the branch from GitHub here and commit my changes?

Reply:
Fantastic! I think you figured it out, github has a nice button to update a PR to current master, makes life super easy ⛱️.

Reply:
Thanks I learned something. I believe indexes are accounts for on startup. We can double check once indexes are added to the dashboard.